tsconfig.json

```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es6",
+    "target": "ES2018",
```

Comment:
This is affecting the produced code's syntax niveau, and hence reduces the set of browsers the clients will work in. Why exactly is this change required? According to https://www.typescriptlang.org/tsconfig/#target , ES6 is still recommended, even for modern browsers.

Since this may be a breaking change for some users, I'd try to avoid this.

Reply:
![error](https://github.com/user-attachments/assets/68478a3a-7d38-43ea-ae6b-d5179f7e7bec)

We'd need to refactor the respective code parts.
I think we already only support a reduced set of browsers by including this in an always called part of `@polar/core`.

🏓 @warm-coolguy

Reply:
I've also just added the type check to the pipeline.
It excludes type errors in clients for now.

Reply:
https://caniuse.com/?search=named%20capture

While named capturing groups are implemented in all modern browsers, other features may be not, and I'm missing a complete list of which browser implements which ES2018 features. For a simple solution, we can @ts-expect-error on this and revert the change to the target.

Reply:
Done 82d4e646be11fae7e64b96cab936323604afd8df

🏓  @warm-coolguy

You can proceed with the review as soon as #149 is merged

Reply:
#149 is merged 🎉